src/core/gen/gen.ml

```diff
@@ -0,0 +1,51 @@
+(* trivial cppo replacement with just enough features to do conditional compilation *)
+
+let if_ocaml_version = Re.(seq [
+    bol; rep blank; str "#if"; rep1 space; str "OCAML_VERSION"; rep1 space; str ">="; rep1 space;
+    char '(';
+    group (rep1 digit); rep1 (set ", "); group (rep1 digit); rep1 (set ", "); group (rep1 digit);
+    char ')'
+  ])
+  |> Re.compile
+
+let version_extract = Re.(seq [
+  group (rep1 digit); char '.'; group (rep1 digit); char '.'; group (rep1 digit)
+  ])
+  |> Re.compile
+
+let current_version =
+  Scanf.sscanf Sys.ocaml_version "%u.%u.%u" (fun a b c -> (a, b, c))
+
+let greater_or_equal (v : (int * int * int)) = current_version >= v
+
+let rec loop output ic =
+  match input_line ic with
+  | line -> (
+    match String.trim line with
+    | "#else" -> loop (not output) ic
+    | "#endif" -> loop true ic
+    | _otherwise -> (
+      match Re.exec_opt if_ocaml_version line with
+      | None ->
+        (* not a condition, it's source code *)
+        if output then
+          print_endline line;
+        loop output ic
+      | Some groups -> (
+        let group = Re.Group.get_opt groups in
+        match group 1, group 2, group 3 with
+        | Some major, Some minor, Some patch ->
+          let major = int_of_string major in
+          let minor = int_of_string minor in
+          let patch = int_of_string patch in
+          let output = greater_or_equal (major, minor, patch) in
+          loop output ic
+        | _ ->
+          loop output ic
```

Comment:
your "error handling" seems to "ignore any errors", which does contradict rules of least surprise. of course for the current state of opamCompat.ml this is fine, but I wonder how hard and complex it now becomes to extend that OCaml modiule, given this ad-hoc script?

Reply:
sorry for the noise, please ignore my comment.